pilgen/src/lib.rs

```diff
@@ -999,4 +1044,56 @@ pol constant p_reg_write_X_CNT = [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] + [0]*;
         let pil = compile::<GoldilocksField>(Some(file_name), &contents).unwrap();
         assert_eq!(format!("{pil}").trim(), expectation.trim());
     }
+
+    #[test]
+    pub fn compile_literal_number_args() {
+        let source = r#"
+reg pc[@pc];
+reg fp;
+
+instr inc_fp amount: unsigned { fp' = fp + amount }
+instr adjust_fp amount: signed, t: label { fp' = fp + amount, pc' = label }
+
+inc_fp 7;
+loop::
+adjust_fp -2, loop;
+"#;
+        let expectation = r#"
+namespace Assembly(1024);
+pol constant first_step = [1] + [0]*;
+pol commit pc;
+(first_step * fp) = 0;
+pol commit fp;
+pol commit instr_inc_fp;
+pol commit instr_inc_fp_param_amount;
+pol commit instr_adjust_fp;
+pol commit instr_adjust_fp_param_amount;
+pol commit instr_adjust_fp_param_t;
+fp' = ((((first_step' * 0) + (instr_inc_fp * (fp + instr_inc_fp_param_amount))) + (instr_adjust_fp * (fp + instr_adjust_fp_param_amount))) + ((1 - ((first_step' + instr_inc_fp) + instr_adjust_fp)) * fp));
+pc' = ((1 - first_step') * ((instr_adjust_fp * label) + ((1 - instr_adjust_fp) * (pc + 1))));
+pol constant p_line = [0, 1, 2] + [2]*;
+pol constant p_instr_adjust_fp = [0, 0, 1] + [1]*;
+pol constant p_instr_adjust_fp_param_amount = [0, 0, -2] + [-2]*;
+pol constant p_instr_adjust_fp_param_t = [0, 0, 1] + [1]*;
+pol constant p_instr_inc_fp = [1, 0, 0] + [0]*;
+pol constant p_instr_inc_fp_param_amount = [7, 0, 0] + [0]*;
+{ pc, instr_inc_fp, instr_inc_fp_param_amount, instr_adjust_fp, instr_adjust_fp_param_amount, instr_adjust_fp_param_t } in { p_line, p_instr_inc_fp, p_instr_inc_fp_param_amount, p_instr_adjust_fp, p_instr_adjust_fp_param_amount, p_instr_adjust_fp_param_t };
+"#;
+        let pil = compile::<GoldilocksField>(None, source).unwrap();
+        assert_eq!(format!("{pil}").trim(), expectation.trim());
+    }
+
+    #[test]
+    #[should_panic]
+    pub fn negative_for_unsigned() {
+        let source = r#"
+reg pc[@pc];
+reg fp;
+
+instr instro x: unsigned { pc' = pc + x }
+
+instro 9223372034707292161;
+"#;
+        compile::<GoldilocksField>(None, source).unwrap();
```

Comment:
What is this testing?

Reply:
The number is so large that it falls into the "negative" part of the field. It fails at this assertion: https://github.com/powdr-org/powdr/pull/310/files#diff-e35bc36cfc5b345185d7dc9a1e99aa31abba8b28dc15bda76e3d658e61881281R385

It's really weird that one cannot test for panic messages in this framework.

Reply:
ah ok